File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def check(s):
    a = int(math.sqrt(s)) + 1
    flag = True
    for i in range(2, a):
        if s % i == 0:
            flag = False
            break
    if s == 1:
        flag = False
    return flag

def numberToBase(n, b):
    if n == 0:
        return 0
    digits = []
    while n:
        digits.append(str(n % b))
        n //= b
    return "".join(digits[::-1])

def solution(n, k):
    new = numberToBase(n, k)
    answer = 0
    tmp = new.split("0")
    for i in tmp:
        if i != "":
            if check(int(i)):
                answer += 1
                

    return answer

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public boolean check(String s) {
        long tmp = Long.parseLong(s);
        boolean flag = true;
        System.out.println(Math.sqrt(tmp));
        for (int i = 2; i <(int) Math.sqrt(tmp) + 1; i++) {
            if (tmp % i == 0) {
                flag = false;
                break;
            }
        }
        if (tmp == 1) {
            flag = false;
        }
        return flag;
    } 
    
    public int solution(int n, int k) {
        String nStr = Integer.toString(n, k);
        String[] strArr = nStr.split("0");
        int answer = 0;
        for (String s : strArr) {
            if (!s.equals("") && check(s)) {
                answer += 1;
                System.out.println(check(s));
            }
        }

        return answer;
    }
}